runtime-attach-distro/src/main/java/io/opentelemetry/contrib/attach/AgentFileProvider.java

```diff
@@ -51,23 +45,18 @@ private static Path createTempDir() {
     return tempDir;
   }
 
-  private static Path createTempAgentJarFile(Path tempDir) {
-    URL url = OpenTelemetryAgent.class.getProtectionDomain().getCodeSource().getLocation();
-    try {
-      return copyTo(url, tempDir, "agent.jar");
+  private Path createTempAgentJarFileIn(Path tempDir) {
+    Path agentJarPath = tempDir.resolve("agent.jar");
```

Comment:
It could be nice. I would rather to keep the code simple and avoid unexpected bugs. Indeed, it would be needed to manage `/` (see [getResourceAsStream javadoc](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getResourceAsStream-java.lang.String-)). Using `URL getResource(String name)` method and `URL` could be problematic, see #359.

Reply:
could using the same filename be problematic if multiple Java apps are using this on the same host (maybe even with different versions of the runtime-attach artifact (and so different versions of the embedded javaagent)?

Reply:
> could using the same filename be problematic if multiple Java apps are using this on the same host (maybe even with different versions of the runtime-attach artifact (and so different versions of the embedded javaagent)?

Each Java app will add a runtime attach dependency; the `agent.jar` will be extracted from this dependency and saved in a temporary folder _different for each Java app_; after that, the `agent.jar` of the temporary folder will be attached to the application. So, each application will use a different agent jar, and I don't see any issue.

Reply:
> saved in a temporary folder _different for each Java app_

oh yes 👍